app/pages/notifications.cjsx

```diff
@@ -0,0 +1,145 @@
+
+  propTypes:
+    project: React.PropTypes.object
+    user: React.PropTypes.object
+
+  getDefaultProps: ->
+    query: page: 1
+
+  getInitialState: ->
+    firstMeta: { }
+    lastMeta: { }
+    notificationsMap: { }
+
+  componentWillMount: ->
+    @getNotifications() if @props.user
+
+  componentWillUnmount: ->
+    @markAsRead('first')()
+    @markAsRead('last')()
+
+  componentWillReceiveProps: (nextProps) ->
+    pageChanged = nextProps.query.page isnt @props.query.page
+    userChanged = nextProps.user and nextProps.user isnt @props.user
+    @getNotifications(nextProps.query.page) if pageChanged or userChanged
+
+  getNotifications: (page = @props.query.page) ->
+    page or= 1
+    query = {page}
+    query.section = "project-#{ @props.project.id }" if @props.project
+    query.section = @props.params.section if @props.params.section
+    talkClient.type('notifications').get(query).then (newNotifications) =>
+      meta = newNotifications[0]?.getMeta() or { }
+      notifications = @state.notifications or newNotifications
+      meta.notificationIds = (n.id for n in newNotifications)
+      notificationsMap = @state.notificationsMap
+
+      for notification in newNotifications
+        notificationsMap[notification.id] = notification
+
+      {firstMeta, lastMeta} = @state
+
+      if meta.page > @state.lastMeta.page
+        lastMeta = meta
+        notifications.push newNotifications...
+      else if meta.page < @state.firstMeta.page
+        firstMeta = meta
+        notifications.unshift newNotifications...
+      else
+        firstMeta = lastMeta = meta
+
+      @setState {notifications, notificationsMap, firstMeta, lastMeta}
+
+  markAsRead: (meta) ->
+    =>
+      ids = @state["#{ meta }Meta"].notificationIds
+      ids = (id for id in ids when not @state.notificationsMap[id].delivered)
+      return if ids.length is 0
+      talkClient.put '/notifications/read', id: ids.join(',')
+      for notification in @state.notifications when notification.id in ids
+        notification.update delivered: true
+
+  markAllAsRead: ->
+    talkClient.put '/notifications/read'
+    for notification in @state.notifications
+      notification.update delivered: true
+    @forceUpdate()
+
+  title: ->
+    if @props.project
+      "#{ @props.project.display_name } Notifications"
+    else if @props.params.section
+      "#{ @props.params.section } Notifications"
+    else
+      'Notifications'
+
+  render: ->
+    <div className="talk notifications">
+      <div className="content-container">
+        <h1 className={"centering title #{ if @props.project then 'talk-module' else '' }"}>
+          {@title()}
+        </h1>
+
+        {if @props.user?
+          <ChangeListener target={@props.user}>{ =>
+            if @state.notifications?.length > 0
+              <div>
+                  <div className="centering">
+                    <div className="talk-module inline-block">
+                      {if @state.firstMeta.page > 1
```

Comment:
Only render the top paginator if we're not starting on the first page of notifications